Rewrite Server Admin Guide (#875)
jamulussoftware/jamulussoftware.github.io: wiki/en/Running-a-Server.md

</details>

## Do I need to run a Server?

> **No**. You can use the Servers listed by the built in Directories and use Jamulus without running a Server yourself. Or you can use a third party hosting service such as [melomax](https://melomax.live/jamulus-hosting/). 

Reviewer: We should not link to a particular Hoster I think. There's Melomax and KOORD, and we should not favour one. 

So give names, but no preference.

Consider using a cloud host, not your home internet connection, to get better ping times if you're having problems.

### General notes

* Any Server should have at least 1.6GHz CPU frequency and 1GB RAM
* Running a Server may require you to adjust any firewalls running on or outside of your machine or cloud host.

> * Running an Unregistered Server at home will require you to port forward on your router. This should not be necessary when running a Registered Server, most cases, but it's advisable to do so because some networks may not work properly with Jamulus in its default mode.

Reviewer: Did we remove the port forward section?

Author: No that section is all in Registered-Servers.md - I'll link to that.

1. Install the Jamulus package for your architecture, for example: 

	`sudo apt install ./{{ site.download_file_names.deb-headless }}`

1. Enable the headless Server process: 

	`sudo systemctl enable jamulus-headless`

1. Add your desired [command line options](Running-a-Server#configuration) to the `ExecStart` line in the systemd service file (by default you will be running an Unregistered Server):

	`sudo systemctl edit --full jamulus-headless`

1. Reload the systemd files and restart the headless Server:

	`sudo systemctl daemon-reload && systemctl restart jamulus-headless` 

_To upgrade your Server, just repeat the steps above._

---

## Server Types

Servers can be run in one of three modes (either at home or on a 3rd party host), depending on your needs.

### 1. Unregistered

This is the default when starting a Server for the first time. Unregistered Servers are not listed by Directories, so only musicians who know your Server's address to will be able to connect to it. This is useful because Jamulus does not let you control who can connect to a Server.

**For information about running an Unregistered Server [see this guide](Unregistered-Servers).**

<figure>

> 	<img src="{% include img/en-screenshots/diagram-unreg-server.inc %}" loading="lazy" alt="Diagram of connections between Clients within a Jamulus Unregistered Server">

Reviewer: Although only slightly related, I think we could/should put these "how things work" into the main repo

Author: Not sure I understand why (and if it means we have to maintain the same information in two places, that's not a good idea really)

</figure>

### 2. Registered

Your Server will be listed in a Directory. Jamulus has a list of built-in Directories that Clients can connect to, and if you register your Server with one of these, anyone can then discover and connect to it. You can also have your Server listed on a Custom Directory (see below), if that better meets your requirements.

Note that Directories can only register up to 150 Servers. If you see a message that says you cannot register your Server becuase the Directory is full, you can try registering on another. For a list of all public Directories and the Servers on them, see [explorer.jamulus.io](https://explorer.jamulus.io).

<figure>
	<img src="{% include img/en-screenshots/diagram-reg-server.inc %}" loading="lazy" alt="Diagram of connections between Clients within a Jamulus Registered Server">
</figure>

### 3. Directory

If you want to run a number of Servers, possibly also behind a firewall or on a LAN, you may want to run your Server as a Directory. Examples include online events, music associations, sectional rehearsals or music lessons for schools.

For information, [see the Directories guide](Directories).

> # Configuration

Reviewer: Probably incorrect heading level. This should not be h1.

Author: Oh actually, it needs to be H1 or the structure doesn't work. Sorry! What's the reason why it shouldn't be an H1?

Reviewer: H1 should be the highest level headline (what the whole page is about): https://yoast.com/one-h1-heading-per-post/

if it’s just about graphics, we should redesign h2 and lower level headings.

Author: It's mainly due to the similar styling I think. Can you remind me which of the .css files we'd need to amend if we did alter these? I always get confused about what gets used for what.

Now you can run these with the `systemctl` command, for example:

`sudo systemctl start jamulusTogglerec` 

You can see the result of these commands if you run `systemctl status jamulus`, or by viewing the logs.

---

## Servers on the desktop

Jamulus can be run in Server mode from the desktop. This gives you a graphical user interface to control most of the settings. To do this, first [install Jamulus](Getting-Started) for your platform, the do one of the following:

* **Windows users** - Use the "Jamulus Server" icon in the Windows Start menu. If you want the Server to start automatically on system start, enable the corresponding check box.

* **macOS users** - Double-click the "Jamulus Server" icon in Applications.

* **Linux users** - Launch the "Jamulus Server" shortcut. Or you can open a terminal window (`CTRL+ALT+t` on Debian and related distros), type `jamulus -s` and hit return.

While most common functions in Jamulus can be set using the GUI, others can only be set using options given in a terminal window when the server is launched. Exactly how you do this will depend on your operating system.

> For example on Windows, to use a specific settings file, right-click on the Jamulus shortcut and choose “Properties” > Target. Add the necessary arguments to Jamulus.exe:

Reviewer 2: Our atypical style guide uses quotes where most technical writing uses **bold**. Alright. Up above, on line 278, we use straight quotes. Here we use the "smart quotes". Which do we prefer here? And in this instance, why is Properties smart-quoted, while Target isn't quoted at all? 

Windows 11 may have made it harder to find the Shortcut option. I had to right-click the EXE icon, choose **Show more options**, and choose **Create shortcut**. Kind of an ordeal! Not sure we should assume Jamulus shortcut is easy to find or make anymore.

Author: You're right. Please suggest some rules on that to add here: https://jamulus.io/contribute/Style-and-Tone - thanks!

Reviewer 2: @pljones expressed a preference for quotation marks, and moving to bold would be a large-scale undertaking.

Reviewer 2: But squealy “smart quotes” are just one dimension of quote hell. What a cross-eyed mess, all squigglin around everywhere. "Properties" > "Target" argh! It's **Properties** -> **Target**. Boom. We should **bold** verbatim interface labels like normal people.

Author: OK so let's go with bold verbatim interface labels then. One for the style guide too :-)

 `"C:\Program Files\Jamulus\Jamulus.exe" --serverbindip 192.168.0.100`

For macOS, start a Terminal window and run Jamulus with the desired options like this:

 `/Applications/Jamulus.app/Contents/MacOS/Jamulus --serverbindip 192.168.0.100`

**Note** Command line options will set the Server’s defaults at startup. You can override them with their corresponding GUI controls while the Server is running.

### The Server status icon

> When a Server is running in GUI mode, the operating system will show an icon in the system tray or status area to represent whether the Server is active:  

Reviewer 2: I can't see these images
```suggestion
When a Server is running in GUI mode, the operating system will show an icon in the system tray or status area that indicates whether the Server is active:  
```

Author: Have you built the site locally? Github won't show them to you in context here.

Reviewer 2: Do you use Sphinx for a local build?

Author: Not as far as I know. You can [use these instructions](https://github.com/jamulussoftware/jamuluswebsite#viewing-the-website-offline). Personally (not sure why), I just git clone, switch the required branch, run the `_po4a-tools/po4a-create-all-targets.sh` script then do a `bundle exec jekyll serve --force_polling`
